### PR TITLE
Upgrade go to 1.21

### PR DESCRIPTION
**Pull request by siddarthkay**
## What does the PR do

- updates status-go reference to build from the branch where we bump go to 1.21 
- https://github.com/status-im/status-go/pull/5216

## Affected areas

Could be many areas since go upgrades impact every feature that communicates with status-go

### CODE REVIEW

File: ci/Dockerfile [hunk 1]
```diff
@@ -81,8 +81,8 @@ RUN wget https://github.com/Kitware/CMake/releases/download/v3.22.1/cmake-3.22.1
 ENV PATH="/opt/cmake/bin:${PATH}"
 
 # Installing Golang
-RUN GOLANG_SHA256="698ef3243972a51ddb4028e4a1ac63dc6d60821bf18e59a807e051fee0a385bd" \
- && GOLANG_TARBALL="go1.20.4.linux-amd64.tar.gz" \
+RUN GOLANG_SHA256="d0398903a16ba2232b389fb31032ddf57cac34efda306a0eebac34f0965a0742" \
+ && GOLANG_TARBALL="go1.21.0.linux-amd64.tar.gz" \
```
Comment (jakubgs) [inline ci/Dockerfile:85]: Please use `1.21.10` to match what we used for CI hosts:

- [`infra-role-golang#5c8083e8`](https://github.com/status-im/infra-role-golang/commit/5c8083e8) - upgrade from 1.20.12 to 1.21.10

And remember to update the image tag for Linux builds.

Comment (siddarthkay) [reply ci/Dockerfile]: Done! Thanks for the feedback.

**Review by jakubgs — CHANGES_REQUESTED** [review]
You are forgetting about Windows.

**Comment by jakubgs** [comment]
The script that manages version for windows is here:


But Scoop package manager that we use never had `1.21.10`, only `1.21.7`, which is better than nothing I guess:

**Comment by jakubgs** [comment]
But it appears Scoop can auto-generate a manifest for `1.21.10`:
```
WARN  Given version (1.21.10) does not match manifest (1.22.3)
WARN  Attempting to generate manifest for 'go' (1.21.10)
Autoupdating go
Searching hash for go1.21.10.windows-386.zip in https://dl.google.com/go/go1.21.10.windows-386.zip.sha256
Found: 0baca8eb50bf36ac92719522dd4b5ca582b5bd01ce98af41a1beab5f78f3f617 using Extract Mode
Searching hash for go1.21.10.windows-amd64.zip in https://dl.google.com/go/go1.21.10.windows-amd64.zip.sha256
Found: 09170b66e7d7c4e2e7a30b8f3350778a8ba5c15951b7eb8ff7545cb86ea9bb71 using Extract Mode
Searching hash for go1.21.10.windows-arm64.zip in https://dl.google.com/go/go1.21.10.windows-arm64.zip.sha256
Found: 163be69c2c289e9acab89ca7f5aa19d780dae121e5f31d06f93d44b028743caf using Extract Mode
Writing updated go manifest
```
So we can use it.

File: ci/Jenkinsfile.windows [hunk 2]
```diff
@@ -5,7 +5,7 @@ library 'status-jenkins-lib@v1.8.13'
 def isPRBuild = utils.isPRBuild()
 
 pipeline {
-  agent { label 'windows && x86_64 && qt-5.15.2' }
+  agent { label 'windows-03 && x86_64 && qt-5.15.2' }
```
[the remark this answers is not in the document]
Comment (siddarthkay) [reply ci/Jenkinsfile.windows]: Done, Thank you.

**Comment by anastasiyaig** [comment]
i ran my full suite for this PR and the results are identical to current master results

**Comment by anastasiyaig** [comment]
Mac m1 build is in forever queue https://ci.status.im/job/status-desktop/job/prs/job/macos/job/aarch64/job/package/job/PR-14946/9/console:

```
> git rev-list --no-walk 34a1ede205dd423cd846e39a76407f5feb4d13c1 # timeout=10
[Pipeline] node
Still waiting to schedule task
There are no nodes with the label ‘[macos&&aarch64&&qt-5.15&&go-1.21](https://ci.status.im/label/macos&&aarch64&&qt-5.15&&go-1.21/)’
```

**Comment by siddarthkay** [comment]
> Mac m1 build is in forever queue

This was the case because we had to apply `go-1.21` change to `ARM` macs and update the label in Jenkins.
We did that for `macm1-01.ih-eu-mda1.ci.devel`

After we merge `status-go` PR we will upgrade other hosts with `go-1.21`

**Comment by anastasiyaig** [comment]
please dont forget :) and thanks for explaining!

**Comment by saledjenic** [comment]
@anastasiyaig please could somebody test keycard flows against this PR, I remember a year ago, that we couldn't change the Go version to 1.21.x, just because we were getting a crash with this from the keycard library I think.
```
signal 16 received but handler not on signal stack
fatal error: non-Go code set up signal handler without SA_ONSTACK flag
```

**Comment by anastasiyaig** [comment]
yes @saledjenic good point, i will do it soon

**Comment by anastasiyaig** [comment]
> @anastasiyaig please could somebody test keycard flows against this PR, I remember a year ago, that we couldn't change the Go version to 1.21.x, just because we were getting a crash with this from the keycard library I think.
> 
> ```
> signal 16 received but handler not on signal stack
> fatal error: non-Go code set up signal handler without SA_ONSTACK flag
> ```

just tried this PR build on my mac with new orange keycard,  and it seem to be working fine ( tried signing transactions, generating accounts, reset keycard)
thanks fore heads up @saledjenic i think its okay

**Review by igor-sirotin — APPROVED** [review]
🎉

**Pull request merged**